napi_object_wrap.md

```diff
@@ -0,0 +1,128 @@
+# N-API Object Wrap APIs
+
+N-API offers a way to "wrap" C++ classes and instances so that they are
+effectively projected into JavaScript.
+
+ 1. The C++ class constructor callback, static properties and methods, and
+    instance properties methods are projected as a JavaScript class using
+    `napi_define_class()`.
+ 2. When JavaScript code invokes the constructor, the constructor callback
+    uses `napi_wrap()` to wrap a new C++ instance in a JavaScript wrapper,
+    then returns the wrapper.
+ 3. When JavaScript code invokes a method or property accessor on the class,
+    the corresponding `napi_callback` is invoked. For an instance callback,
+    `napi_unwrap()` is used to obtain the C++ instance that is the target of
+    the call.
+
+### *napi_define_class*
+
+#### Signature
+```C
+napi_status napi_define_class(napi_env env,
+                              const char* utf8name,
+                              napi_callback constructor,
+                              void* data,
+                              size_t property_count,
+                              const napi_property_descriptor* properties,
+                              napi_value* result);
+```
+
+#### Parameters
+ - `[in]  env`: The environment that the API is invoked under
+ - `[in]  utf8name`: Name of the JavaScript constructor function; this is
+   not required to be the same as the C++ class name, though it is recommended
+   for clarity
+ - `[in]  constructor`: Callback function that handles constructing instances
+   of the class. (This should be a static method on the class, not an actual
+   C++ constructor function.)
+ - `[in]  data`: Optional data to be passed to the constructor callback as
+   the `data' property of the callback info
+ - `[in]  property_count`: Number of items in the `properties` array argument
+ - `[in]  properties`: Array of property descriptors describing static and
+   instance data properties, accessors, and methods on the class
+   See `napi_property_descriptor`
+ - `[out] result`: A `napi_value` representing the constructor function for
+   the class
+
+#### Description
+Projects a C++ class to JavaScript. Specifically, the C++ constructor callback
```

Comment:
The "Projects" terminology is not resonating with me.  Would something like "Binds" work ?